src/registrar/registrar_middleware.py

```diff
@@ -60,13 +68,17 @@ def process_view(self, request, view_func, view_args, view_kwargs):
             return None
 
         if request.user.is_authenticated:
-            if hasattr(request.user, "finished_setup") and not request.user.finished_setup:
-                return self._handle_setup_not_finished(request)
+            if request.user.verification_type == User.VerificationTypeChoices.REGULAR:
+                if hasattr(request.user, "finished_setup") and not request.user.finished_setup:
+                    return self._handle_regular_user_setup_not_finished(request)
+            else:
+                if hasattr(request.user, "finished_setup") and not request.user.finished_setup:
+                    return self._handle_other_user_setup_not_finished(request)
```

Comment:
@zandercymatics I don't know if my input will be useful here but adding that initially this design was born out of a need to block invited domain managers who bypass IV from being in our system without contact information.  I think, in the case of users added to the Verified by staff table, it also makes sense to put them in this flow because we still need contact information from them. I know there might be some other cases where users IAL1 so I'm unsure of those use cases. Do you see any problems here?

Reply:
ahhhh I see. Sorry if I missed that i your PR Zander.